test/utils.sh

```diff
@@ -2,49 +2,53 @@
 # set -o pipefail
 
 detect_zenroom_path() {
-	zenroom_paths=( "$PWD" "$PWD/../../src" "$PWD/../src" "$PWD/src"
-					"/usr/local/bin" "/usr/bin" "/bin" "$PWD/../.." "$PWD/..")
-	zenroom_path="/usr/local/bin/zenroom"
-	case $OSTYPE in
-		linux*)
-			zenroom_name="zenroom-linux-amd64"
-			case `uname -m` in
-				i686) zenroom_name="zenroom-linux-i386" ;;
-				i386) zenroom_name="zenroom-linux-i386" ;;
-				arm*) zenroom_name="zenroom-linux-armhf" ;;
-				aarch64) zenroom_name="zenroom-linux-aarch64" ;;
-			esac
-			for p in "${zenroom_paths[@]}"; do
-				if test -r "$p/$zenroom_name"; then zenroom_path="$p/$zenroom_name"; break; fi
-				if test -r "$p/zenroom"; then zenroom_path="$p/zenroom"; break; fi
-			done
-			unset zenroom_name
-		;;
-		darwin*)
-			for p in "${zenroom_paths[@]}"; do
-				if test -r "$p/zenroom-osx.command"; then zenroom_path="$p/zenroom-osx.command"; break; fi
-				if test -r "$p/zenroom.command"; then zenroom_path="$p/zenroom.command"; break; fi
-				if test -r "$p/zenroom"; then zenroom_path="$p/zenroom"; break; fi
-			done
-		;;
-		cygwin*|msys*|win32*)
-			for p in "${zenroom_paths[@]}"; do
-				if test -r "$p/zenroom.exe"; then zenroom_path="$p/zenroom.exe"; break; fi
-			done
-		;;
-		bsd*)
-			for p in "${zenroom_paths[@]}"; do
-				if test -r "$p/zenroom"; then zenroom_path="$p/zenroom"; break; fi
-			done
-		;;
-	esac
+	if test "$is_cortexm" == true; then
```

Comment:
If you move this `if` statement right after switch-esac statement, you can minimize the proposed changes with fewer lines.